Update HIBP sensor to use API v3 and API Key

aetaric opened the pull request:
## Breaking Change:
HaveIBeenPwned API now requires an API key. Said key has a USD$3.50 monthly charge associated.

## Description:
HaveIBeenPwned api v2 is being deprecated and API keys are now required. For more info on this: https://www.troyhunt.com/authentication-and-the-have-i-been-pwned-api/
This PR moves the sensor to use V3 and to use the user provided API key.

**Related issue (if applicable):** fixes #25388

**Pull request with documentation for [home-assistant.io](https://github.com/home-assistant/home-assistant.io) (if applicable):** home-assistant/home-assistant.io#9977

## Example entry for `configuration.yaml` (if applicable):
```yaml
sensor:
  - platform: haveibeenpwned
    email:
      - your_email1@domain.com
      - your_email2@domain.com
    api_key: apikeygoeshere
```

## Checklist:
  - [x] The code change is tested and works locally.
  - [x] Local tests pass with `tox`. **Your PR cannot be merged unless tests pass**
  - [x] There is no commented out code in this PR.
  - [x] I have followed the [development checklist][dev-checklist]

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [home-assistant.io](https://github.com/home-assistant/home-assistant.io)

If the code communicates with devices, web services, or third-party tools:
  - [x] [_The manifest file_][manifest-docs] has all fields filled out correctly. Update and include derived files by running `python3 -m script.hassfest`.
  - [x] New or updated dependencies have been added to `requirements_all.txt` by running `python3 -m script.gen_requirements_all`.
  - [x] Untested files have been added to `.coveragerc`.

If the code does not interact with devices:
  - [x] Tests have been added to verify that the new code works.

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html

cgtobi committed:
Fix Netatmo climate battery level (#25165)

* Interpolate battery level

* Sort list

balloob committed:
Updated frontend to 20190721.1

zxdavb committed:
Tweak evohome migration (#25281)

* Initial commit

add hvac_action to zones, remove target_temp from controller

fix incorrect hvac_action

de-lint

Initial commit

de-lint & minor refactor

tweak docstrings, and type hints

tweak docstrings

* refactor setpoints property

* tweak docstring

* tweak docstring

* avoid a unnecessary I/O

* avoid unnecessary I/O

* refactor schedule/setpoints

* remove type hint

* remove type hint 2

* tweak code

* delint type hints

* fix regression

amelchio committed:
Update pysonos to 0.0.22 (#25399)

zxdavb committed:
[climate] Correct evohome hvac_action (#25407)

* inital commit

* take account of rounding up of curr_temp

zxdavb committed:
[climate] Bugfix/Tweak honeywell migration (#25369)

* refactor supported_features

add dr_data

delint

simplify code

refactor target_temps, and delinting

delint

fix typo

add min/max temps

delint

refactor target temps

refactor target temps 2

correct typo

tweak

fix potential bug

fix potential bug 2

bugfix entity_id

fix typo

* remove explicit entity_id

* refactor hvac_action

* refactor hvac_action

* bugfix - HVAC_MODE_HEAT_COOL incorrectly added to hvac_modes

cgtobi committed:
Bump pyatmo version to 2.1.2 (#25296)

fredrike committed:
Update Daikin preset modes (#25395)

* fix preset documentation

* Use pydaikin set holiday method

balloob committed:
Bumped version to 0.96.4

balloob committed:
Merge pull request #25452 from home-assistant/rc

0.96.4

zxdavb committed:
Bump geniushub client (#25458)

balloob committed:
Fix Nest turning off eco (#25472)

* Fix Nest turning off eco

* Add hvac action

balloob committed:
Allow cors for static files (#25468)

balloob committed:
Bumped version to 0.96.5

balloob committed:
Merge pull request #25486 from home-assistant/rc

0.96.5

aetaric committed:
Update HIBP sensor to use API v3 and API Key

HaveIBeenPwned api v2 is being deprecated and API keys are now required. For more info on this: https://www.troyhunt.com/authentication-and-the-have-i-been-pwned-api/

aetaric committed:
Update HIBP sensor to use API v3 and API Key

HaveIBeenPwned api v2 is being deprecated and API keys are now required. For more info on this: https://www.troyhunt.com/authentication-and-the-have-i-been-pwned-api/

MartinHjelmare commented on homeassistant/components/haveibeenpwned/sensor.py:
```diff
@@ -142,13 +146,11 @@ def update_no_throttle(self):
-            req = requests.get(
-                url, headers={USER_AGENT: HA_USER_AGENT}, allow_redirects=True,
-                timeout=5)
+            req = requests.get(url, headers=header,
```
It would be good to switch to a pypi package. I haven't found one that supports api version 3.

aetaric replied:
While I agree a pypi package is probably a good idea in most use cases, in this case it's a single API call. Installing and loading an entire library for a single API call feels a bit heavy handed to me personally.

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

squishykid committed:
Update solax to 0.1.2 (#25497)

bollewolle committed:
Add De Lijn (Flemish Public Transport) component (#24265)

* Initial commit of De Lijn (Flemish Public Transport) component

* Code corrections as per HA dev's requests

* changes to variable naming, setting attribution and states, plus some smaller optimizations

* Overlooked some linting issues, these are now fixed

* Updated pydelijn version requirement to 0.5.1 so UTC timestamps can be used instead of relative/local times.
Removed unused definition

* Updated pydelijn version requirement to 0.5.1 in requirements_all.txt

* Update the self._attributes dict directly instead of replacing it
Assign ATTRIBUTION while creating the _attributes dict
Remove the ATTRIBUTION assignment in device_state_attributes as it's updated in the async_update now.

* Linting issue (lenght of 2 lines) solved

* Removed a relative time attribute
Updated a linting issue in the LOGGER (used % instead of the format)

pvizeli committed:
[skip ci] Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

balloob committed:
Fix python 3.5 test

balloob committed:
Fix deprecation warning in test (#25506)

cgarwood committed:
Improve handling of Z-Wave config entry vs yaml config (#25112)

* Improved handling of config entry vs yaml config

* Address review comment

balloob committed:
Fire lovelace updated event when update detected (#25507)

gjbadros committed:
Support multiple Elk instances (#23839)

* Support multiple Elk instances

* Allow more than one Elk M1 alarm system to be integrated into a single hass instance.

* Introduces new "devices" schema at the top level, each of which has
  the prior configuration schema.

* Requires new version of elkm1, 0.7.14, that gwww and I just updated (thanks Glen!)

QUESTION: Should the "devices" section be optional to avoid breaking
old configuration files?  I chose not to do that for simplicity and
because I was following the doorbird code which requires the "devices"
section for all configurations even with only one device.

* Fixed a bunch of hound-raised issues

Fixed issues raised by hound -- there was clearly
a tool I was supposed to run to get those warnings
before submitting the PR.  Sorry!

Updated REQUIREMENTS.

* Fixed whitespace and line-length mistakes

Also fixed unused prefix local variable lint warning.

* Fixed missing blank line

* Fixed more lint warnings.

Not sure if I missed these on the first pass or if the linter stopped
after a certain number of warnings or something else.

Switched logging to use %d and %s instead of string concatenation (per
lint request and because I imagine it migth be better performing
in some (oldish, I presume) implementations of python.

* Fixed typo in last commit.

* Eliminate devices subsection in config schema

This eliminates the breaking change for configurations wanting a
singleton elk m1 instance (the majority of users, no doubt).  I did
not do it like this before because I was following the lead of the
doorbird component which introduced a devices: section when moving
to support multiple doorbells.  But Rohan Kapoor kindly pointed me
at the zoneminder component which sets the other (IMO) preferable
precedent. Will update the docs change shortly.

* Call async_add_entities once for all the elk controllers.

Just move async_add_entities() outside of the loops across the elk m1
controllers, so it's called once for each platform.

* Call async_add_entities only once per platform.

Move it to after the loop, so it's called only once
per platform even when there are multiple elk m1 controllers.

* Various improvements to be more idiomatic python + bug fixes

Thanks to Martin Hjelmare for the careful review and suggestions.
(All mistaken improvements and new bugs are my own.)

* Removed semicolon that lint caught.

* Idiomatic python improvements

Use dict.values() (instead of making it easier to add local looping variable
on the keys by using _, bar = ...items())

Use [] when the key is known to exist.

* Support multiple Elk instances

* Allow more than one Elk M1 alarm system to be integrated into a single hass instance.

* Introduces new "devices" schema at the top level, each of which has
  the prior configuration schema.

* Requires new version of elkm1, 0.7.14, that gwww and I just updated (thanks Glen!)

QUESTION: Should the "devices" section be optional to avoid breaking
old configuration files?  I chose not to do that for simplicity and
because I was following the doorbird code which requires the "devices"
section for all configurations even with only one device.

* Fixed a bunch of hound-raised issues

Fixed issues raised by hound -- there was clearly
a tool I was supposed to run to get those warnings
before submitting the PR.  Sorry!

Updated REQUIREMENTS.

* Fixed whitespace and line-length mistakes

Also fixed unused prefix local variable lint warning.

* Fixed missing blank line

* Fixed more lint warnings.

Not sure if I missed these on the first pass or if the linter stopped
after a certain number of warnings or something else.

Switched logging to use %d and %s instead of string concatenation (per
lint request and because I imagine it migth be better performing
in some (oldish, I presume) implementations of python.

* Fixed typo in last commit.

* Eliminate devices subsection in config schema

This eliminates the breaking change for configurations wanting a
singleton elk m1 instance (the majority of users, no doubt).  I did
not do it like this before because I was following the lead of the
doorbird component which introduced a devices: section when moving
to support multiple doorbells.  But Rohan Kapoor kindly pointed me
at the zoneminder component which sets the other (IMO) preferable
precedent. Will update the docs change shortly.

* Call async_add_entities once for all the elk controllers.

Just move async_add_entities() outside of the loops across the elk m1
controllers, so it's called once for each platform.

* Call async_add_entities only once per platform.

Move it to after the loop, so it's called only once
per platform even when there are multiple elk m1 controllers.

* Various improvements to be more idiomatic python + bug fixes

Thanks to Martin Hjelmare for the careful review and suggestions.
(All mistaken improvements and new bugs are my own.)

* Removed semicolon that lint caught.

* Idiomatic python improvements

Use dict.values() (instead of making it easier to add local looping variable
on the keys by using _, bar = ...items())

Use [] when the key is known to exist.

* Use dict[key] instead of .get (incl. fixing typo). Use .values() instead of .items() when ignoring keys.

* Gotta use devices.get(prefix) since we use no prefix for the singleton elk instance

* fix requirement to use newer elkm1 that supports my changes for multiple elk devices

* Removed spurious + between a string broken between two lines for formatting; was failing a lint check about logging needing to use %s

* Remove REQUIREMENTS and DEPENDENCIES since those are now taken care of by the manifest.json file.

* Add configuration check that the prefixes are all unique

* Use new dependency 'getmac' to get mac address of Elk M1 controllers and use that for uniqueid if possible, else use None.  Also removed some procedural checking of unique prefix since that's now handled at schema check time.

* Whitespace changes to make style checker happy and code more consistent

* Removed unused variable, added blank line

* Make getmac a requirement not dependency

I should've RTFM.

* ws only change; I really need to get Emacs to understand these style guidelines

* Ran script/gen_requirements_all.py; script/setup needed to be run so that was failing.

* More style check fixes and one bug fix.

* Incomplete set of changes from last push

* More conform-to-hass-style changes: use caps to start log message (and do not use function name even for debug message. And do not use string concatenation; prefer new-style .format.

* Style fixes.

* Switch back to using the prefix config field for setting the unique_id since the mac address approach has numerous shortcomings including: 1) new dependency; 2) lack of reliability; 3) doesn't work for serial connections; 4) breaks when a layer 4+ networking entity intermediates the elk m1 connection.

* Reran to update (removing getmac dependency)

* Skipped trailing ','; keep forgetting which languages are forgiving about this practical nicety of allowing trailing commas without changing the semantics.

* Validate uniqueness on lowercase versions of the prefix since we're gonna use .lower() on creating the entity id that has to be unique; do the _has_all_unique_prefixes check last so we get errors from the device schema before complaining about the uniqueness problem, if any

* Use vol.Lower to convert to lowercase instead of the map.  Also fixed a pair of bugs for the alarm control panel display message service -- since data templates always generate strings, the values subject to range/set restrictions need to be coerced to their proper type before the check

* Fix some flake8 warnings.

* Fixed typo; it's Coerce not coerce.

* Use elkm1m_ string to start unique_id when and only when there is a non-empty prefix given; this enables backward compatibility to avoid a breaking change by letting the elkm1_ start to unique_id keep working exactly as it used to.

* minor comment tweak to force automation tests to run again since they failed for unrelated reasons last time

* There's actually been a 0.7.15 release which was meta-information and tidying only so we might as well depend on it

* Forgot to update this with gen_requirements_all.py

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

oncleben31 committed:
Add container settings for YAML extension to avoid Hass specific custom tags errors (#25504)

* Add settings for YAML extension to avoid !secret tag errors

* add other HA custom tags.

manonstreet committed:
Add last_run_success boolean attribute to Switchbot for error trapping (#25474)

* Add last_run_error boolean attribute to Switchbot entity to allow for trapping of errors

* Add last_run_success boolean attribute to Switchbot for error trapping

* Add last_run_success boolean attribute to Switchbot for error trapping

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

Kane610 committed:
deCONZ - Add power attribute for consumption sensors (#25512)

* Add power attribute for consumption sensors

* Bump dependency to v62

thomasgermain committed:
Improve seventeentrack (#25454)

* Code improvement + tests

* review

* review + moving to pytest test function

* move test to async

* remove code comment

Anglac committed:
Roombalocate (#25508)

* Add Roomba Locate

* Update vacuum.py

Kane610 committed:
deCONZ - cleanup sensor attributes (#25540)

* Improve attributes

mjj4791 committed:
Update to buienradar json api; and additional monitored_conditions  (#24463)

* Update to json api; and additional sensors

* remove unneeded coordinate rounding

* minor optimizations

* Update CODEOWNERS

* Update sensor.py

unit conversion for:
- windgust (ms to km/h)
- windspeed (m/s to km/h)
- windspeed_1d (ms/km/h)
- visibility (m to km)

* Update weather.py

unit conversion for windspeed (m/s to km/h)

* Update sensor.py

* Update weather.py

* Update weather.py

* Update CODEOWNERS

* Update manifest.json

* Update CODEOWNERS

* Update CODEOWNERS

yet another try to get CODEOWNERS accepted...

* update codeowners

* update codeowners

* update

* updated codeowners to match manifest

bachya committed:
Fix bug with WWLLN update interval (#25498)

* Fix bug with WWLLN update interval

* Match the docs

cameronrmorris committed:
Fix WinkAC mode API calls to correct methods (#25545)

Commit:
Make cool_on and heat_on method calls. They aren't properties in python-wink (#25549)

michaeldavie committed:
Bump env_canada to 0.0.19 (#25548)

* Bump env_canada to 0.0.19

* Update requirements_all.txt

Danielhiversen committed:
Tibber, off peak values (#25320)

* Tibber, off peak values

* style

* style

* refactor

* style

cereal2nd committed:
Enable velbus config entries (#25308)

* Initial work on config_flow

* Finish config flow

* Pylint checks, make sure the import only happens once

* Added support for unloading, small fixes

* Check in the hassfest output files

* Flake8 fixes

* pylint mistake after flake8 fixes

* Work on comments

* Abort the import if it is already imported

* More comments resolved

* Added testcases for velbus config flow

* Fix pylint and flake8

* Added connection test to the config flow

* More sugestions

* renamed the abort reason

* excluded all but the config_flow.py from the velbus component in coveragerc

* Rewrote testcases with a patched version of _test_connection

* Docstyle fixes

* Updated the velbus testcases

* just yield

* flake8 fixes

scop committed:
huawei_lte: try unsupported data retrievals only once (#25524)

* huawei_lte: try unsupported data retrievals only once

Refs https://github.com/home-assistant/home-assistant/pull/23809

* Move huawei_lte_api imports to top level

scop committed:
Ignore .dmypy.json (#25528)

amelchio committed:
Update eternalegypt to 0.0.8 (#25551)

fbradyirl committed:
Update Cisco Mobility Express module version (#25422)

* Update requirements_all.txt

* Update manifest.json

* Bump to remove f-strings

* Bump to remove f-strings

Commit:
Expose last_video_id as property for Ring camera (#25553)

andersonshatch committed:
Quiet noisy tado query logging (#25529)

* Quiet noisy tado query logging

* Lint fix

ktnrg45 committed:
Add PS4 tests for media player (#25415)

* Add tests for media_player

* remove ps4/media_player.py

* Add unsubscribe method to unload

* Add_to_hass instead of add_to_manager

* Use hass.states for states

* Fix assertions

* fix tests

* Add schedule update

* Remove entity assertions

Kane610 committed:
UniFi POE control restore clients (#25558)

* Restore POE controls on restart

alengwenus committed:
LCN cover control via output ports (#25511)

* LCN motor control via oputput ports

* Remove default value from cover validator

tofuSCHNITZEL committed:
updated telegram trusted IPs (#25564)

updated telegram trusted IPs (Source: https://core.telegram.org/bots/webhooks)

Kane610 committed:
UniFi device tracker restore clients (#25532)

bachya committed:
Transition SimpliSafe data retrieval to its own object (#25546)

* Transition SimpliSafe data retrieval to its own object

* Don't overwrite a variable

* Member comments

* Member comments

Kane610 committed:
UniFi - Track devices (#25570)

JonGilmore committed:
pylutron PyPI update (#25557)

* pylutron PyPI update

We've been working with the original maintainer of pylutron, and they've published an update to PyPI to support a couple different things: homeowner keypads, main repeater keypads

* added requirements

cereal2nd committed:
Make the velbus component more robust in handling errors (#25567)

* Add some try excepts for velbus as the python-velbus lib is not good in handling these

* Only catch velbusExceptions

* only wrap the lines that can cause the exception

* Fix indentation mixup

orson1282 committed:
Updated Workday Binary Sensor to use Holidays 0.9.11 and added support for Aruba Holidays. (#25568)

* Updated Workday Binary Sensor to use Holidays 0.9.11 and added support for Aruba Holidays.

* updated requirements_all.txts

benleb committed:
add cleaning state code for roborock s6 (#25500)

Adminiuga committed:
ZHA log helper (#25543)

* Logging helper.
* Use log helper for ZHA entities.
* Use log helper for ZHA core device.
* Log helper for ZHA core channels.
* Lint
* ZHA fixture fix.

Quentame committed:
Fix : Velbus translation error (#25575)

balloob committed:
Fix Ecobee HVAC action + available presets (#25488)

* Fix Ecobee HVAC action + available presets

* Update python-ecobee-api to 0.0.21

* Include proper operation list.

* Allows pass on preset to set_climate_hold

* Remove aux heat as a preset

* Fix test

bachya committed:
Add last event data (including "changed_by") to SimpliSafe (#25569)

* Add "last event" sensor for SimpliSafe

* Functionality round 1

* Cleanup

* Whitespace

* Whitespace

* Updated requirements

* Removed unused constants

* Member comments

balloob committed:
Bump Python support to min Python 3.6.0 (#25582)

* Bump Python support to min Python 3.6.0

* Fix type

balloob committed:
Add Black

teharris1 committed:
Bump insteonplm to 0.16.5 (#25580)

JonGilmore committed:
Fix status of lutron switches/lights after HA reboot (#25592)

manutenfruits committed:
Add support for Roku TVs to be powered on or off (#25590)

michaeldavie committed:
Bump env_canada to 0.0.20 (#25594)

Commit:
Bump venstarcolortouch to 0.9 (#25585)

* Update manifest.json

* Update requirements_all.txt

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

pvizeli committed:
Update azure-pipelines-ci.yml for Azure Pipelines

elupus committed:
Log platform import errors and correct reqs for config check (#25425)

* Log failures to load plaforms

* Drop unused exception variable

* Also load pip requirements with check config

* Fix lint

* More lint

* Drop invalid parameters for error call

DarkFox committed:
Improve and align Rejseplanen with other transport components (Breaking) (#25375)

* Improve and align Rejseplanen with other transport components (Breaking)

* Fix empty list check

* Remove pointless list cast

* Clean up redundant definition of transport types

* Add docstring

* Simplify _times check

bachya committed:
Add migration notification for Ambient PWS (#25561)

* Add migration notification for Ambient PWS

* Remove unnecessary kwarg

Co-Authored-By: Andrew Sayre <6730289+andrewsayre@users.noreply.github.com>

* Automatically recreate config entry

* Delete entities and devices

* Move imports

croghostrider committed:
Fix wrong exposed light for emulated hue  (#25581)

* Add auto detect if brightness is supported

* Fix

* Fix tests

* Cleanup

zxdavb committed:
Fix bug and bump geniushub client (#25599)

Fix bug, delint and bump client

tofuSCHNITZEL committed:
implemented timout setting for telnet switch (#25602)

ChristianKuehnel committed:
remove myself from CODEOWNDERS (#25593)

* remove myself from CODEOWNDERS

Sorry, I do not have time right now to work on HomeAssistant, so I'm removing myself from the CODEOWNERS.

* Update manifest.json

rossdargan committed:
Change how ring polls for changes to allow more platforms to be added (#25534)

* Add in a switch to control lights and sirens

* Improve the way sensors are updated

* fixes following flake8

* remove light platform, and fix breaking test.

* Resolve issues with tests

* add tests for the switch platform

* fix up flake8 errors

* fix the long strings

* fix naming on private method.

* updates following p/r

* further fixes following pr

* removed import

* add additional tests to improve code coverage

* forgot to check this in

yeralin committed:
Introduce support for color temperature (#25503)

* Introduce support for color temperature

* Fix linter errors

* Remove extra whitespace for pylint

cgtobi committed:
Netatmo climate refactor (#25457)

* Refactor climate component to use home id rather than name

* Bump pyatmo version

* Add new exception

* Update pyatmo version

balloob committed:
Bump frontend to 20190731.0

thomasloven committed:
Return history for entities in the order they were requested (#25560)

* Return history for entities in the order they were requested

* Fix problems. Add tests

* Update __init__.py

scop committed:
Python < 3.6 remainder cleanups (#25607)

scop committed:
Azure flake8 dep, docstring fixes (#25605)

* Fix Azure CI flake8 deps

* Docstyle fixes

balloob committed:
Black

balloob committed:
Lint

balloob committed:
Lint

balloob committed:
Type

balloob committed:
Make sure typing matches stdlib

balloob committed:
Uninstall typing

balloob committed:
Confirm deletion

balloob committed:
Add constraints to key files

balloob committed:
Merge pull request #25584 from home-assistant/black

Black

balloob committed:
Add translations

balloob committed:
Merge remote-tracking branch 'origin/master' into dev

balloob committed:
Version bump to 0.98.0.dev0

dmulcahey committed:
bump quirks (#25618)

scop committed:
script/test: fix tox env (#25620)

scop committed:
Move mypy settings to setup.cfg (#25611)

pvizeli committed:
Remove liner for editor. Not need with black

tleegaard committed:
Add support for HomeKit CO2 sensors (#25603)

elupus committed:
Fix test_install_existing_package (#25627)

homeassistant.util.package.install_package is not
a corutine.

Kane610 committed:
Handle disabled devices (#25625)

scop committed:
Blacken top level *.py (#25621)

* Blacken top level *.py

* Tolerate double quotes too in setup.py dependency extraction

Jc2k committed:
Fix polling HomeKit devices with multiple services per accessory (#25629)

scop committed:
Huawei LTE sensor unique id improvements (#25609)

* Convert sensor setup to async

* Improve sensor unique ids

* Save some indent levels, use f-string formatting

* Require getmac in tests

* Fix RouterData init in tests

* Make discovery_info optional in async_setup_platform signature

balloob committed:
Filter out empty results in history API (#25633)

balloob committed:
Expose comfort presets as HA presets (#25491)

* Expose comfort presets as HA presets

* Fix bugs

* Handle unavailable

* log level debug on update

* Lint

scop committed:
Add some debugging to azure mypy job (#25632)

Jc2k committed:
Bump homekit_python to 0.15 (#25631)

oncleben31 committed:
Meteofrance improve log error messages (#25630)

* Improve log error messages

* remove unique_id not ready yet

DarkFox committed:
Fix handling of empty results from Rejseplanen (#25610)

* Improve handling of empty results from Rejseplanen (Fixes #25566)

* Exclude attributes with null value

* Add period back into docstring

* Fix formatting

balloob committed:
Upgrade hass-nabucasa to 0.16 (#25636)

nielstron committed:
Add each fronius sensor as own template (#25608)

* Fix formatting in history test

* Add each sensor as own template

* Make adapters non-entities

* Externalize and organize data fetching, improve system view

* Small fixes

Rename fetching adapters to adapters
throw away non-working system overviews
slightly change naming
remove scan_interval from schema
formatting

* Scan interval is already timedelta and unnecessary return

* Formatting

* Ensure better codestyle by storing cell variables explicitely in different places

CoMPaTech committed:
Add Plugwise component (#25533)

* Added Plugwise component

* pylint fixes

* Additional pylints (local tox not noticing these)

* Changes according to review

* CI flake8 correction

* Applying Black

* Review changes, without exception and schema

* Review changes, exception improvement

* Review changes, exception correctio and schema cleanup

* Further cleaning as per review

* Removed blank line

* Add debugging to API init

* Add debugging to API init without dot

balloob committed:
Updated frontend to 20190801.0

balloob committed:
Add preset to be away and eco (#25643)

Commit:
Add HmIP-SCI to Homematic IP Cloud, Fix HmIP-SWDM (#25639)

* Add HmIP-SCI to Homematic IP Cloud

* Bump upstream dependency

* Fix HmIP-SWDM

cereal2nd committed:
Fix velbus codeowner (#25649)

Santobert committed:
Feature zwave preset modes (#25537)

* Initial commit

* Add some more code

* Local tests passing

* Remove unnecessary line

* Add preset attributes to __init__

* Remove some more debugger lines

* Add some tests

* Fix comparision to None

* Improve test coverage

* Use unknown modes as presets

* Bugfixes and test improvements

* Add tests for unknown preset modes

* linting

* Improve mappings

* Move PRESET_MANUFACTURER_SPECIFIC to zwave

* Replace isinstance with cast

* Add test for hvac_action

* hvac_mode is never None

* Improved mapping of current mode to hvac/preset modes

* Fix bugs where hvac_mode is None

* Add default hvac mode

* Fixed default hvac mode

* Fix linting

* Make flake happy

* Another linting

* Make black happy

* Complete list of default hvac modes

* Add mapping to heat/cool eco

* Fixed another bug where mapping goes wrong

Kane610 committed:
UniFi - allow configuration to not track clients or devices (#25642)

* Allow configuration to not track clients or devices

Adminiuga committed:
ZHA code cleanup. (#25644)

* isort ZHA imports.

* Sort zha channel registry.

* Sort ZHA core registry.

* Sort ZHA core consts.

johnnychicago committed:
Add ELECTRICITY_IMPORTED_TOTAL obis to dsmr (#25655)

Adminiuga committed:
isort ZHA imports. (#25660)

scop committed:
Azure mypy related tweaks (#25663)

* Revert "Add some debugging to azure mypy job (#25632)"

This reverts commit 767b8e9f25affbebb98d65f57c0d78b532a951ba.

No longer needed.

* Install setup.py dependencies for mypy in Azure

For better coverage, and to match what tox does.

amelchio committed:
Revert flux_led to 0.89 (#25653)

* Revert Black

* Revert "Introduce support for color temperature (#25503)"

This reverts commit e1d884a4841f853bb4247a76fca417c0b4f4e7c1.

* Revert "Fix flux_led only-white controllers (#22210)"

This reverts commit 48138189b3c24261fe62a78b6ec854c761d5ce63.

* Revert "Fix MagicHome LEDs with flux_led component (#20733)"

This reverts commit 1444a684e02fab99648f6e5daea9f28b6cf45c10.

* Re-Black

* Use mode detection for scanned bulbs

scop committed:
Spelling fixes (#25666)

Kane610 committed:
Options to not track wired clients (#25669)

Commit:
Add PRESET_AWAY to HomematicIP Cloud climate (#25641)

* enable climate away_mode and home.refresh

* Add Party eco modes

aetaric committed:
Fix requested changes

Resolves issues in PR home-assistant/home-assistant#25495

aetaric committed:
Fix requested changes

Resolves issues in PR home-assistant/home-assistant#25495

MartinHjelmare commented on homeassistant/components/haveibeenpwned/sensor.py:
```diff
@@ -25,17 +25,19 @@
 
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_EMAIL): vol.All(cv.ensure_list, [cv.string]),
+    vol.Required(CONF_API_KEY): vol.All(cv.string),
```
Remove `vol.All`. That's only needed if more than one validator should be combined.

MartinHjelmare commented:
Please rebaseon latest  dev branch and solve the merge conflicts.

pvizeli committed:
Azure ci templates (#25688)

* Use templates for Azure-CI steps

* Migrate ci

* fix endpoint

Kane610 committed:
In some circumstances device.last_seen can be None (#25690)

Kane610 committed:
UniFi - reverse connectivity logic (#25691)

* Make connectivity control in line with other implementations

pvizeli committed:
Add schedules to yml

pvizeli committed:
Merge branch 'dev' of https://github.com/home-assistant/home-assistant into dev

scop committed:
Upgrade pydocstyle to 4.0.0, do not run in tox (#25667)

* Upgrade pydocstyle to 4.0.0 and flake8-docstrings to 1.3.1

http://www.pydocstyle.org/en/4.0.0/release_notes.html#july-6th-2019

* Address pydocstyle D413's

* tox: do not run pydocstyle

Does not seem to add any value over flake8-docstrings (and would have
needed a D202 exclusion).

CoMPaTech committed:
Plugwise haanna 0.10.1 (#25693)

* Update module dependency (py3.6 fix)

* Update module dependency (py3.6 fix) - update requirements

aetaric committed:
fix verification to not call vol.All since it's not needed

aetaric committed:
Rebase dev

aetaric commented:
Rebased, but github things 🙃 . Other issue is resolved.

aetaric committed:
accidently a blank line

MartinHjelmare commented:
The branch has unrelated commits after the rebase. It's ok to open a new PR, and link to this one, if you don't want to take the scenic git tour.

aetaric commented:
Will do. Closing and will open a new one.
